src/transformStateWithClones.js

```diff
@@ -7,7 +7,60 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateArr = [];
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+    const newState = stateArr.length
+      ? { ...stateArr[stateArr.length - 1] }
+      : { ...state };
```

Comment:
The variable name `newState` does not fully adhere to the naming conventions suggested in the checklist. Consider renaming it to be more descriptive, such as `stateCopy` to indicate that it is a copy of the current state. [NAMING]